Source/ASDisplayNode+Layout.mm

```diff
@@ -352,8 +352,10 @@ - (void)_locked_measureNodeWithBoundsIfNecessary:(CGRect)bounds
     ASLayout *layout = [self calculateLayoutThatFits:constrainedSize
                                     restrictedToSize:self.style.size
                                 relativeToParentSize:boundsSizeForLayout];
-    
     nextLayout = std::make_shared<ASDisplayNodeLayout>(layout, constrainedSize, boundsSizeForLayout, version);
+    // Now that the constrained size of pending layout might have been reused, the layout is useless
+    // Release it and any orphaned subnodes it retains
+    _pendingDisplayNodeLayout = nullptr;
```

Comment:
It is true that in case the pending layout can't be applied because of a bounds mismatch, we can keep it around. I'm fine with adding another flag [here](https://github.com/TextureGroup/Texture/pull/455/files#diff-71ea742a29543cb7e08c3dffe2a2508eR340) to handle that edge.

However, as said above, in case `_pendingDisplayNodeLayout->version < _layoutVersion`, it's quite important to get rid of it.